src/main/java/storage/WriteDataFiles.java

```diff
@@ -40,33 +50,33 @@ public void executeFunction() {
      * a string.
      */
     private void storeCheatSheet() {
-        /*for(CheatSheet cheatSheet : cheatSheets) {
+        for (CheatSheet cheatSheet : cheatSheets) {
             convertStringToFile(cheatSheet);
         }
-        */
     }
 
     /**
      * Converts the cheatSheet into a file format.
      *
      * @param cheatSheet The cheatSheet that is currently being converted into a file.
      */
-    private void convertStringToFile(CheatSheet cheatSheet) {
+    public void convertStringToFile(CheatSheet cheatSheet) {
+        Path textFileDirectory;
         StringBuilder cheatSheetFileBuild = new StringBuilder();
 
-        // build cheatsheet content
+        // Build cheatsheet content
         buildFileContents(cheatSheetFileBuild, cheatSheet);
 
-        String fileName = cheatSheet.getCheatSheetName() + FILE_EXTENSION;
-        Path textFileDirectory = Paths.get(USER_DIR, DATA, fileName);
+        String fileName = cheatSheet.getCheatSheetName();
+        textFileDirectory = (debugPath == null) ? Paths.get(USER_DIR, DATA, fileName) : debugPath;
 
         try {
```

Comment:
it seems like for both read and write classes will create Files if not present already, perhaps make It only the job of write to create both the folders and files